src/jit/decomposelongs.cpp

```diff
@@ -920,15 +925,24 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
 
     GenTree* tree   = use.Def();
     GenTree* gtLong = tree->gtGetOp1();
+    GenTree* oldShiftOp = tree->gtGetOp2();
+
     genTreeOps oper = tree->OperGet();
+    genTreeOps shiftOper = oldShiftOp->OperGet();
+
```

Comment:
and shiftWidthOp. How about shiftCountOper, shiftCountUse, oldShiftCountOp, shiftCountOp, etc?

---
In reply to: [78654147](https://github.com/dotnet/coreclr/pull/7169#discussion_r78654147) [](ancestors = 78654147,78653812)